code/AssetLib/glTF2/glTF2Importer.cpp

```diff
@@ -453,6 +453,12 @@ void glTF2Importer::ImportMeshes(glTF2::Asset &r) {
     unsigned int k = 0;
     meshOffsets.clear();
 
+
+    const unsigned int unusedIndex = ~0;
+    std::vector<unsigned int> usedVertexIndices;
+    std::vector<unsigned int> reverseMappingIndices;
```

Comment:
For containers sizes that are unknown upfront, a deque usually performs better, since it never needs to reallocate existing data. Both reverseMappingIndices and indexBuffer don't seem to be needed to be contiguous, you only access individual elements by index, so they could be deque's, unless you have a good approximate size to reserve them with up front.

Reply:
Good point, I'll consider this

Reply:
Ah, now I remember: While I do like the ideas of deque's, the std version of a deque is completely unusable imho, because it has a ridiculously small buffer size. So leaving it like this. BTW, the reallocation concern is the reason why I declared arrays outside both loops, so it reuses pre-warmed arrays

Reply:
The deque implementation of some major compiler vendor(s) are debatable, indeed. The no-realloc argument is valid anyway. Obviously there is a trade-off point where either std::vector or std::deque is better, and without concrete testing it is really hard to say which one is better. Since you already measured a major speedup with the current solution, I guess it's good enough.